docs/src/components/header/HeaderNavBar.tsx

```diff
@@ -190,38 +190,61 @@ export default function HeaderNavBar() {
       });
     }
     if (event.key === 'Escape') {
-      setSubMenuOpen(false);
+      setSubMenuOpen(null);
       setSubMenuIndex(null);
     }
   }
+
+  const setSubMenuOpenDebounced = React.useMemo(
+    () => debounce(setSubMenuOpen, 40),
+    [setSubMenuOpen],
+  );
+
+  const setSubMenuOpenUndebounce = React.useMemo(
+    () => (value: typeof subMenuOpen) => {
+      setSubMenuOpenDebounced.clear();
+      setSubMenuOpen(value);
+    },
+    [setSubMenuOpen, setSubMenuOpenDebounced],
+  );
+
+  React.useEffect(() => {
+    return () => {
+      setSubMenuOpenDebounced.clear();
+    };
+  }, [setSubMenuOpenDebounced]);
+
   return (
     <Navigation>
       <ul ref={navRef} role="menubar" onKeyDown={handleLeftRightArrow}>
         {FEATURE_TOGGLE.nav_products && (
           <li
             role="none"
-            onMouseOver={() => setSubMenuOpen(true)}
-            onFocus={() => setSubMenuOpen(true)}
-            onMouseOut={() => setSubMenuOpen(false)}
-            onBlur={() => setSubMenuOpen(false)}
+            onMouseEnter={() => setSubMenuOpenUndebounce('products')}
+            onFocus={() => setSubMenuOpenUndebounce('products')}
+            onMouseLeave={() => setSubMenuOpenDebounced(null)}
+            onBlur={() => setSubMenuOpenUndebounce(null)}
           >
             <div
               role="menuitem"
               tabIndex={0}
               id="products-menu"
               ref={productsMenuRef}
               aria-haspopup
-              aria-expanded={subMenuOpen ? 'true' : 'false'}
+              aria-expanded={subMenuOpen === 'products' ? 'true' : 'false'}
               onKeyDown={handleKeyDown}
             >
               Products
             </div>
             <Popper
-              open={subMenuOpen}
+              open={subMenuOpen === 'products'}
               anchorEl={productsMenuRef.current}
               transition
               placement="bottom-start"
-              style={{ zIndex: 1200, pointerEvents: subMenuOpen ? 'visible' : 'none' }}
+              style={{
+                zIndex: 1200,
+                pointerEvents: subMenuOpen === 'products' ? undefined : 'none',
```

Comment:
The default value of `pointer-events ` is `'auto'` not `'visible'`. Also `visible` only makes sense in the context of an SVG. Instead, we can simply unset the value.